### PR TITLE
First pass at dashboard user logging

### PR DESCRIPTION
**Pull request by hardillb**
## Description

Explain how to track which users have viewed a Dashboard v2 page.

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        "wires": []
+    }
+]
+```
+
+Next we need to add the SVG to the Dashboard, this can be done by adding a Template node with the following HTML content.
+
+```html
+<div>
+    <object data="/tracker" height="1" width="1"></object>
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:153]: Nice, didn't even know this was possible. TIL 😄

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    - flowfuse
+---
+
+As we continue to add features to the Node-RED Dashboard v2 one feature request that came in was to track which users had visited a Dashboard. Multi user support for the Dashboard is on the backlog but this could be solved with the parts that are currently available.
+
+<!--more-->
+
+## FlowFuse Authentication
+
+One of the features we off on FlowFuse is the ability to protect HTTP endpoints and Dashboards using the same FlowFuse user authentication that protects access to the FlowFuse Application and the Node-RED instances.
```
Comment (knolleary) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:20]:
```suggestion
One of the features we offer on FlowFuse is the ability to protect HTTP endpoints and Dashboards using the same FlowFuse user authentication that protects access to the FlowFuse Application and the Node-RED instances.
```

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+We even offer a specific RBAC role that just allows access to these endpoints but not the FlowFuse application.
```
Comment (knolleary) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:22]:
```suggestion
We even offer a specific RBAC 'viewer' role that just allows access to these endpoints but not the FlowFuse application.
```

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+FlowFuse authentication can be enabled from the Projects Settings page on the Security tab
```
Comment (knolleary) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:24]:
```suggestion
FlowFuse authentication can be enabled from the Instance Settings page on the Security tab
```

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+## Implementation
+
+First we will create a HTTP-in/HTTP-response pair to serve up a single pixel SVG image. I chose SVG as it doesn't require creating a binary image file to load.
+
+The following flow snippet includes both the HTTP-in/HTTP-response nodes and a change node to set the `msg.payload` to the SVG content and to set the HTTP headers to include the correct mime type.
+
+There is also a second change node which extracts the user information.
+
+```json
```
Comment (knolleary) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:38]: Can I suggest we minify the JSON?

**Review by knolleary — COMMENTED** [review]
Some minor comments and tweaks suggested.

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
+---
+title: "Tracking Who Has Opened a Dashboard"
+subtitle: "Using FlowFuse Authentication Audit Dashboard v2 Access"
+description: "Using FlowFuse Authentication Audit Dashboard v2 Access"
+date: 2023-11-09
+authors: ["ben-hardill"]
+image: "/blog/2023/11/images/dashboard-user-flow.png"
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:7]: While this works, it creates a fuzzy image for me

File: src/blog/2023/11/dashboard-2.0-user-tracking.md (hunk 5 above)
Comment (ZJvandeWeg) [reply src/blog/2023/11/dashboard-2.0-user-tracking.md:38]: I cannot make the multi line comment of that many lines, but the JSON is


```
[{"id":"7f22dc81d8192d4d","type":"http in","z":"98c8d7ea66149291","name":"","url":"/tracker","method":"get","upload":false,"swaggerDoc":"","x":210,"y":460,"wires":[["7d36739c02cd04ec","5f4647c97917cce1"]]},{"id":"58fd30516a077e29","type":"http response","z":"98c8d7ea66149291","name":"","statusCode":"","headers":{},"x":630,"y":460,"wires":[]},{"id":"7d36739c02cd04ec","type":"change","z":"98c8d7ea66149291","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"<svg width=\"1\" height=\"1\"> <rect width=\"1\" height=\"1\" style=\"fill:rgb(255,255,255);stroke-width:3;stroke:rgb(0,0,0)\" /> Sorry, your browser does not support inline SVG.</svg>","tot":"str"},{"t":"set","p":"headers","pt":"msg","to":"{\"Content-Type\":\"image/svg+xml\"}","tot":"json"}],"action":"","property":"","from":"","to":"","reg":false,"x":420,"y":460,"wires":[["58fd30516a077e29"]]},{"id":"5f4647c97917cce1","type":"change","z":"98c8d7ea66149291","name":"","rules":[{"t":"set","p":"payload","pt":"msg","to":"req.session.user","tot":"msg"}],"action":"","property":"","from":"","to":"","reg":false,"x":420,"y":520,"wires":[["ddc02b4e9c30c807"]]},{"id":"ddc02b4e9c30c807","type":"debug","z":"98c8d7ea66149291","name":"debug 2","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":600,"y":520,"wires":[]}]
```

**Review by zackwasli — APPROVED** [review]

File: src/blog/2023/11/dashboard-2.0-user-tracking.md (hunk 6 above)
Comment (ZJvandeWeg) [reply src/blog/2023/11/dashboard-2.0-user-tracking.md:7]: I think you're correct, in DEV_MODE locally it yields different results compared to the Netlify deploy preview. Sorry about this, the image is fine indeed.

File: src/blog/2023/11/dashboard-2.0-user-tracking.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
+---
+title: "Tracking Who Has Opened a Dashboard"
+subtitle: "Using FlowFuse Authentication Audit Dashboard v2 Access"
+description: "Using FlowFuse Authentication Audit Dashboard v2 Access"
+date: 2023-11-09
```
Comment (ZJvandeWeg) [inline src/blog/2023/11/dashboard-2.0-user-tracking.md:5]:
```suggestion
date: 2023-11-13
```

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**